src/dmd/dinterpret.d

```diff
@@ -836,6 +1012,53 @@ private Expression interpret(FuncDeclaration fd, InterState* istate, Expressions
         }
         eargs[i] = earg;
     }
+ 
+    // Skip generated symbols and only permit caching of top-level CTFE calls
+    bool cacheable = (istate == null || istate.caller == null)
```

Comment:
How about having `cacheable` be off if `global.params.cache` is off? I.e. why test both repeatedly in subsequent code?